packages/transformers/src/modules/state_restore/schedule/schedule_transformer.ts

```diff
@@ -0,0 +1,53 @@
+import { XMessage } from "@samagra-x/xmessage";
+import { ITransformer } from "../../common/transformer.interface";
+
+export class ScheduleTransformer implements ITransformer {
+
+    /// Accepted config properties:
+    ///     timerDuration: number: (required) Specifies the duration of the timer in milliseconds. This value is mandatory and 
+    ///         determines how long the timer will run before it triggers the next action.
+    ///     immediateRestore: boolean: (optional, default: false) A flag indicating whether the state should be restored 
+    ///         immediately after clearing an existing timer. If set to `true`, the transformer's state is reset to the state
+    ///         stored in `metaData.restoreState` after clearing the timer. 
+    
+    private timers: Map<string, NodeJS.Timeout> = new Map();
+
+    constructor(readonly config: Record<string, any>) { }
+
+    async transform(xmsg: XMessage): Promise<XMessage> {
+        console.log(`SCHEDULE_TRANSFORMER called.`);
+
+        if (!this.config.timerDuration) {
+            throw new Error('timerDuration is required!');
+        }
+
+        const timerId = `${xmsg.from.userID}_timer`;
```

Comment:
use a better unique identifier to prevent collisions from multiple providers that have similar userId, for example, phone number.
```suggestion
        const timerId = `timer_${xmsg.from.userID}_${xmsg.channelId}_${xmsg.providerId}`;
```